grakn-graql/src/main/java/ai/grakn/graql/internal/parser/QueryParser.java

```diff
@@ -199,6 +246,34 @@ public Pattern parsePattern(String patternString){
         return visit.apply(getQueryVisitor(), tree);
     }
 
+    /**
+     * Consume a single query from the given token stream.
+     *
+     * @param tokenStream the {@link TokenStream} to consume
+     * @return a new {@link TokenSource} containing the tokens comprising the query
+     */
+    private TokenSource consumeOneQuery(TokenStream tokenStream) {
+        List<Token> tokens = new ArrayList<>();
+
+        boolean startedQuery = false;
+
+        while (true) {
+            Token token = tokenStream.LT(1);
+            boolean isNewQuery = NEW_QUERY_TOKENS.contains(token.getType());
```

Comment:
Could you clarify the logic here. This many booleans makes it a bit hard to read.